### PR TITLE
ci: Modify Firecracker to install from repository

### LINKED ISSUE #1091: ci: Modify firecracker to install from repository
Instead of installing firecracker, runtime, and all the other Kata
components from the tar, this will change to install from the sources.

### PR DESCRIPTION
**Pull request by GabyCT**
Instead of installing firecracker, runtime, and all the other Kata
components from the tar, this will change to install from the sources.

Fixes #1091

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: .ci/install_firecracker.sh [hunk 1]
```diff
@@ -26,17 +27,23 @@ if [ "$docker_version" != "18.06" ]; then
...
+firecracker_version=$(get_version "assets.hypervisor.firecracker.version")
+[ -n "$firecracker_version" ] || die "failed to get firecracker version"
+
+# Get firecracker
+go get -d ${firecracker_repo} || true
+# Checkout to specific version
+pushd "${GOPATH}/src/${firecracker_repo}"
+git checkout tags/${firecracker_version}
+./tools/devtool --unattended build --release -- --features vsock
+sudo mv ${GOPATH}/src/${firecracker_repo}/build/release/firecracker /usr/bin/
```
Comment (chavafg) [inline .ci/install_firecracker.sh]: use `install` instead?

File: .ci/setup.sh [hunk 2]
```diff
@@ -121,6 +116,13 @@ install_extra_tools() {
 		echo "Openshift not installed"
 }
 
+modify_firecracker_configuration_toml() {
```
Comment (chavafg) [inline .ci/setup.sh]: can this be handled at the end inside `.ci/install_runtime.sh`?

File: .ci/setup.sh [hunk 3]
```diff
@@ -121,6 +116,13 @@ install_extra_tools() {
...
+	runtime_path="/usr/share/defaults/kata-containers"
+	echo "Enable firecracker configuration.toml"
+	sudo mv ${runtime_path}/configuration-fc.toml ${runtime_path}/configuration.toml
+	sudo systemctl daemon-reload
```
Comment (chavafg) [inline .ci/setup.sh]: we shouldn't need to run this line

**Comment by GabyCT** [comment]
@chavafg changes applied

File: .ci/install_firecracker.sh [hunk 4]
```diff
@@ -26,17 +27,23 @@ if [ "$docker_version" != "18.06" ]; then
...
+# Get version for firecracker from runtime/versions.yaml
+firecracker_version=$(get_version "assets.hypervisor.firecracker.version")
+[ -n "$firecracker_version" ] || die "failed to get firecracker version"
+
+# Get firecracker
+go get -d ${firecracker_repo} || true
+# Checkout to specific version
+pushd "${GOPATH}/src/${firecracker_repo}"
+git checkout tags/${firecracker_version}
+./tools/devtool --unattended build --release -- --features vsock
```
Comment (jcvenegas) [inline .ci/install_firecracker.sh]: good, I had the idea you'd reuse the script on https://github.com/kata-containers/packaging/blob/master/static-build/firecracker/build-static-firecracker.sh to not duplicated code, but if this wont change a lot lets keep in this way. 
@mcastelino  @jodh-intel  any precerence here?

Comment (jcvenegas) [inline .ci/install_firecracker.sh]: good, I had the idea you'd reuse the script on https://github.com/kata-containers/packaging/blob/master/static-build/firecracker/build-static-firecracker.sh to not duplicated code, but if this wont change a lot lets keep in this way. 
@mcastelino  @jodh-intel  any precerence here?

**Comment by GabyCT** [comment]
@grahamwhaley , `metrics-ci` is failing with `memory-footprint` as it says that is exceeding memory, any objection with this change?

**Comment by grahamwhaley** [comment]
haha - I just opened https://github.com/kata-containers/ci/issues/102
the fail is v.likely not to do with this PR

File: .ci/install_runtime.sh [hunk 5]
```diff
@@ -77,7 +78,17 @@ if [ "$USE_VSOCK" == "yes" ]; then
 	fi
 fi
 
-echo "Add runtime as a new/default Docker runtime. Docker version \"$(docker --version)\" could change according to updates."
-docker_options="-D --add-runtime kata-runtime=/usr/local/bin/kata-runtime"
-echo "Add kata-runtime as a new/default Docker runtime."
-"${cidir}/../cmd/container-manager/manage_ctr_mgr.sh" docker configure -r kata-runtime -f
+if [ "$KATA_HYPERVISOR" == "qemu" ]; then
```
Comment (mcastelino) [inline .ci/install_runtime.sh]: @GabyCT looks like we made it either or. Can we support both at the same time, potentially one of them via a shell script. I will work on eliminating the shell script based on 
https://github.com/kata-containers/runtime/issues/1082

Till then can we route fc through a shell script

File: .ci/install_firecracker.sh (hunk 4 above)
Comment (mcastelino) [reply .ci/install_firecracker.sh]: @jcvenegas I am ok with a little duplication as long the right version gets picked up. I am planning to update to the latest version today or next week (to 0.14)

**Pull request merged**